DocumentsFromSnapshotMigration/src/main/java/com/rfs/RfsMigrateDocuments.java

```diff
@@ -208,13 +231,15 @@ public static DocumentsRunner.CompletionStatus run(Function<Path,LuceneDocuments
                     }
                     return shardMetadata;
                 },
-                unpackerFactory, readerFactory, reindexer).migrateNextShard();
+                unpackerFactory, readerFactory, reindexer)
+                .migrateNextShard(rootDocumentContext::createReindexContext);
```

Comment:
The documents runner to have a context passed into it, no?

Also the way `DocumentsRunner` is written is hard to understand, lets reduce the inlining